### PR TITLE
Add multiple AuthPlugin support and some fixes.

### PR DESCRIPTION
**Pull request by yoh1496**
This PR contains below.

### Multiple AuthPlugin Support

To load multiple instances by one class, `AuthPluginLoader` interface is added.
By implementing AuthPluginLoader containing `loadInstances()` , you can load multiple AuthPlugin onto Personium unit.

### some fixes

- modifing headers.
- Replace CachingHttpClient with CachingHttpClientBuilder.
- If properties file is not found, throw exception.

### CODE REVIEW

File: src/main/java/io/personium/plugin/base/auth/AuthPluginLoader.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+/**
+ * Interface of providing multiple AuthPlugin instance.
+ */
+public interface AuthPluginLoader {
+    /**
+     * Instanciate Multiple AuthPlugin.
+     * @return ArrayList of AuthPlugin
+     */
+    ArrayList<AuthPlugin> loadInstances();
```
Comment (tochi-y) [inline src/main/java/io/personium/plugin/base/auth/AuthPluginLoader.java]:
```suggestion
    List<AuthPlugin> loadInstances();
```

File: src/main/java/io/personium/plugin/base/auth/AuthPluginLoader.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+import java.util.ArrayList;
+
+/**
+ * Interface of providing multiple AuthPlugin instance.
+ */
+public interface AuthPluginLoader {
+    /**
+     * Instanciate Multiple AuthPlugin.
+     * @return ArrayList of AuthPlugin
```
Comment (tochi-y) [inline src/main/java/io/personium/plugin/base/auth/AuthPluginLoader.java]:
```suggestion
     * @return List of AuthPlugin
```

**Review by tochi-y — COMMENTED** [review]
@yoh1496 I added comments. Please check them, thanks.

File: src/main/java/io/personium/plugin/base/auth/AuthPluginLoader.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.personium.plugin.base.auth;
+
+import java.util.ArrayList;
```
Comment (tochi-y) [inline src/main/java/io/personium/plugin/base/auth/AuthPluginLoader.java]:
```suggestion
import java.util.List;
```

**Comment by yoh1496** [comment]
Thank you for merging!

**Pull request merged**